### PR TITLE
CI update

### LINKED ISSUE #120: CI Update for GitHub actions
**Describe the bug**
GitHub actions is deprecating an old node.  To fix this, we need to repace:

```
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
```

with

```
    - uses: actions/checkout@v3
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v4
```

In the .yml files.

**To Reproduce**
Go to the actions and read the warnings: https://github.com/pysat/pysat/actions/runs/3283879182

**Expected behavior**
No warnings in the actions

**Desktop (please complete the following information):**
 - OS: GitHub Actions

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #120 by updating the CI.  Also addresses: https://github.com/orgs/pysat/projects/2

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Must be tested on GitHub Actions.

**Test Configuration**:
* Operating system: GitHub Actions
* Version number: Python 3.8, 3.9, 3.10
* Any details about your local setup that are relevant: main pysat

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,28 +17,29 @@ jobs:
...
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-
-    - name: Install standard dependencies
-      run: |
-        pip install -r test_requirements.txt
-        pip install -r requirements.txt
+        pip install "xarray<2022.06"
```
Comment (jklenzing) [inline .github/workflows/main.yml]: either delete or move to the NEP29 module

Comment (aburrell) [inline .github/workflows/main.yml]:
```suggestion
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -17,28 +17,29 @@ jobs:
...
 
     - name: Install NEP29 dependencies
       if: ${{ matrix.numpy_ver != 'latest'}}
       run: |
         pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
```
Comment (aburrell) [inline .github/workflows/main.yml]:
```suggestion
        pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
        pip install "xarray<2022.06"
```

**Review by jklenzing — COMMENTED** [review]
From the GA log for the nep29 job:
```
Installing collected packages: pytz, six, pyparsing, numpy, python-dateutil, packaging, pandas, xarray
  Attempting uninstall: numpy
    Found existing installation: numpy 1.20.0
    Uninstalling numpy-1.20.0:
      Successfully uninstalled numpy-1.20.0
Successfully installed numpy-1.23.4 packaging-21.3 pandas-1.5.1 pyparsing-3.0.9 python-dateutil-2.8.2 pytz-2022.5 six-1.16.0 xarray-2022.3.0
```
I'm not sure why it's updating numpy after installing the requested version.  I have a new structure up at https://github.com/sami2py/sami2py/pull/175 that fixes the issue.

Sorry about the confusion on this update.  I think I was looking at the wrong part of the log before.

**Comment by aburrell** [comment]
Ok @jklenzing what do you think?  The alternative is to limit the pandas version to one that works with numpy 1.20.0 (probably 1.4.x).

**Comment by jklenzing** [comment]
> Ok @jklenzing what do you think? The alternative is to limit the pandas version to one that works with numpy 1.20.0 (probably 1.4.x).

I'm happy enough with this.  In January, numpy 1.21 becomes the new minimum, so it becomes moot at that point.  I think we need to discuss how we proceed across pysat packages at the next meeting. At the moment, I'm leaning toward the separate requirements file like in sami2py.  Probably the right way is to implement docker at some point.

**Review by jklenzing — APPROVED** [review]

**Pull request merged**